### PR TITLE
feat: discord bot demo

### PR DESCRIPTION
**Pull request by Fyko**
This PR:
- creates a demo discord bot using Supabase

TODO:
- [x] testing
- [x] docs

### CODE REVIEW

File: discord-bot-demo/backend/supabase/config.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# A string used to distinguish different Supabase projects on the same host. Defaults to the working
+# directory name when running `supabase init`.
+project_id = "viewer-polls-supabase-example"
```
Comment (rileymiller) [inline discord-bot-demo/backend/supabase/config.toml:3]: I'd update the project_id to discord-bot-example

File: discord-bot-demo/backend/supabase/functions/interactions/user.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import {
+  APIChatInputApplicationCommandInteraction,
+  APIEmbed,
+  Snowflake,
+} from "https://deno.land/x/discord_api_types@0.36.1/v10.ts";
+import { stripIndents } from "https://deno.land/x/deno_tags@1.8.0/tags/stripIndents.ts";
+import { getDiscordConnection } from "../util/truffle/index.ts";
+import { createResponse } from "../util/respond.ts";
+
+const orgId = Deno.env.get("TRUFFLE_ORG_ID")!;
```
Comment (rileymiller) [inline discord-bot-demo/backend/supabase/functions/interactions/user.ts:10]: We can probably go ahead and add the code now to grab the orgId from the discordGuildId, we can run a script on `truffle-dev-early-access` to add the Connection in the interim until we build out the admin side to add Discord guild's

File: discord-bot-demo/backend/supabase/config.toml (hunk 1 above)
Comment (Fyko) [reply discord-bot-demo/backend/supabase/config.toml]: impl in https://github.com/trufflehq/truffle-packages/pull/8/commits/b02c18d62d112b4eed2c6f29ab494c36832ad2f2

File: discord-bot-demo/deno.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "tasks": {
+        "deploy-pkg": "truffle-cli deploy",
+        "deploy-fn": "cd backend/ && supabase functions deploy viewer-polls-example-handler --no-verify-jwt",
```
Comment (rileymiller) [inline discord-bot-demo/deno.json:4]: will need to update this to the new supabase function

File: discord-bot-demo/backend/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+* Create a poll in Truffle
+
+### Getting Started
+To get setup on the Supabase side of things you can follow the guide from the Supabase official [docs](https://supabase.com/docs/guides/functions#creating-a-function).
+
+* Install the `supabase` CLI. [Docs](https://supabase.com/docs/reference/cli/installing-and-updating)
+* Login to the CLI `supabase login`. [Docs](https://supabase.com/docs/reference/cli/supabase-login)
+* You can skip the `supabase init` step and repurpose the `supabase/` folder and `config.toml` file from the forked `@truffle/events-demo-backend` package. If you skip `supabase init` you will need to create a project through the [Supabase Admin Dashboard](https://app.supabase.com/) and update the `project_id` field inside of the `config.toml`.
+* Link to your remote Supabase project with `supabase link --project-ref <ref>`. You can find your project ref if you click on the project in the Supabase Admin and grab the ref from the url e.g `https://app.supabase.com/project/<ref>` [Docs](https://supabase.com/docs/reference/cli/supabase-link) 
+* Deploy the Supabase Edge Function `supabase functions deploy viewer-polls-example-handler --no-verify-jwt`
```
Comment (rileymiller) [inline discord-bot-demo/backend/README.md:15]: update to the new function

File: discord-bot-demo/backend/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+* Receive webhooks when the Viewer Poll collectible is redeemed
+* Lookup the user who redeemed the collectible
+* Create a poll in Truffle
+
+### Getting Started
+To get setup on the Supabase side of things you can follow the guide from the Supabase official [docs](https://supabase.com/docs/guides/functions#creating-a-function).
+
+* Install the `supabase` CLI. [Docs](https://supabase.com/docs/reference/cli/installing-and-updating)
+* Login to the CLI `supabase login`. [Docs](https://supabase.com/docs/reference/cli/supabase-login)
+* You can skip the `supabase init` step and repurpose the `supabase/` folder and `config.toml` file from the forked `@truffle/events-demo-backend` package. If you skip `supabase init` you will need to create a project through the [Supabase Admin Dashboard](https://app.supabase.com/) and update the `project_id` field inside of the `config.toml`.
```
Comment (rileymiller) [inline discord-bot-demo/backend/README.md:13]: update the truffle package name

File: discord-bot-demo/backend/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Truffle Webhook Supabase Edge Function Example
+
+`backend/supabase/functions/viewer-polls-example-handler` is an example of a Supabase Edge Function written in Deno that will:
```
Comment (rileymiller) [inline discord-bot-demo/backend/README.md:3]: update the function name

**Review by rileymiller — COMMENTED** [review]
We'll also want to set this up as a truffle package w/ a truffle config and use the package api key instead of the org api key. Not super necessary for this specific example but when folks eventually need to access a privileged endpoint that requires permissions the bot will need to use the package api key instead of the org api key so the package has permissions for each truffle org it's installed on

**Review by rileymiller — APPROVED** [review]

**Pull request merged**